Cython/Utils.py

```diff
@@ -54,8 +57,28 @@ def wrapper(*args):
     return wrapper
 
 
+def method_and_cache_names(instance):
+    """Yields the method name and cache name,
+    the cache name already exists in the instance,
+    whereas the method name may be missing"""
+
+    for attr_name in dir(instance):
+        match = _METHOD_CACHE_PATTERN.match(attr_name)
+        if match is None:
+            continue
+
+        yield match.group(1), attr_name
```

Comment:
This function seems overkill to me. Definitely as a public function (without leading `_`).
And almost half of it is a docstring that I don't even understand.
I'd rather keep this in the `clear_method_cache()` function to make it clearer what that does.

I'm also not all that happy with the constants. I understand that they keep both strings closer to each other. But the code that uses them doesn't seem to benefit from the constants.

Reply:
> But the code that uses them doesn't seem to benefit from the constants.

Wdym?

Also they are used in tests, not like we couldn't copy strings to tests, but constants seems to be a bit clearer.